From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fabian=20Gr=C3=BCnbichler?= <f.gruenbichler@proxmox.com>
Date: Tue, 7 Mar 2017 13:29:31 +0100
Subject: [PATCH] add Provides/Replaces: corosync-pve
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Fabian Grünbichler <f.gruenbichler@proxmox.com>
---
 debian/control | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/debian/control b/debian/control
index fad56353..d11dc18c 100644
--- a/debian/control
+++ b/debian/control
@@ -31,6 +31,11 @@ Homepage: http://corosync.github.io/corosync/
 Vcs-Git: https://anonscm.debian.org/git/debian-ha/corosync.git
 Vcs-Browser: https://anonscm.debian.org/cgit/debian-ha/corosync.git
 
+Package: corosync-pve
+Architecture: all
+Depends: corosync (= ${binary:Version})
+Description: Transitional package.
+
 Package: corosync
 Architecture: any
 Pre-Depends: ${misc:Pre-Depends}
@@ -39,6 +44,8 @@ Depends: ${shlibs:Depends},
  lsb-base (>= 3.0-6),
  adduser,
  xsltproc
+Provides: corosync-pve (= ${binary:Version})
+Replaces: corosync-pve (<< 2.4.2-2)
 Description: cluster engine daemon and utilities
  ${DESCRIPTION}
  This package contains the Corosync daemon and some administration tools.
-- 
2.14.2

