From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fabian=20Gr=C3=BCnbichler?= <f.gruenbichler@proxmox.com>
Date: Wed, 8 Mar 2017 12:12:05 +0100
Subject: [PATCH] restart after package upgrade is complete
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Fabian Grünbichler <f.gruenbichler@proxmox.com>
---
 debian/rules | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/debian/rules b/debian/rules
index d43bf29b..f45847cc 100755
--- a/debian/rules
+++ b/debian/rules
@@ -86,6 +86,9 @@ override_dh_systemd_enable:
 	dh_systemd_enable -p corosync-qdevice --no-enable
 	dh_systemd_enable --remaining-packages
 
+override_dh_systemd_start:
+	dh_systemd_start -r --restart-after-upgrade
+
 override_dh_strip:
 	dh_strip --dbgsym-migration=corosync-dbg
 
-- 
2.14.2

