From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fabian=20Gr=C3=BCnbichler?= <f.gruenbichler@proxmox.com>
Date: Mon, 13 Mar 2017 13:26:29 +0100
Subject: [PATCH] add libcorosync4-pve transitional package
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Fabian Grünbichler <f.gruenbichler@proxmox.com>
---
 debian/control | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/debian/control b/debian/control
index 8a2ed29e..15e1cf7e 100644
--- a/debian/control
+++ b/debian/control
@@ -49,6 +49,11 @@ Architecture: all
 Depends: corosync (= ${binary:Version})
 Description: Transitional package.
 
+Package: libcorosync4-pve
+Architecture: all
+Depends: corosync (= ${binary:Version})
+Description: Transitional package.
+
 Package: corosync
 Architecture: any
 Pre-Depends: ${misc:Pre-Depends}
